pkg/input/promread/promread.go

```diff
@@ -0,0 +1,234 @@
+package promread
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"github.com/thanos-community/obslytics/pkg/version"
+
+	"time"
+)
+
+type promReadInput struct {
```

Comment:
```suggestion
type Input struct {
```
Why not this? (:

Reply:
I think we should keep it consistent with other Input types and create a new PR to refactor both of them at the same time

Reply:
wdyt?